app/controllers/drivers_controller.rb

```diff
@@ -0,0 +1,66 @@
+class DriversController < ApplicationController
+
+  def index
+    @drivers = Driver.all
+  end
+
+  def new
+    @driver = Driver.new
+  end
+
+  def create
+    @driver = Driver.new(driver_params)
+
+    if @driver.save
+      redirect_to drivers_path
+    else
+      render :new, status: :bad_request
+    end
+  end
+
+  def edit
+    @driver = Driver.find_by(id: params[:id])
+    if @driver.nil?
+      head :not_found
+    end
+  end
+
+  def update
+    @driver = Driver.find(params[:id])
+
+    if @driver.nil?
+      head :not_found
+    end
+
+    is_updated = @driver.update(driver_params)
+
+    if is_updated
+      redirect_to driver_trips_path(@driver)
+    else
+      render :edit, status: :bad_request
+    end
+  end
+
+  def destroy
+    driver = Driver.find_by(id: params[:id])
+
+    if driver.nil?
+      head :not_found
+    end
+
+    if driver.destroy
+      redirect_to drivers_path
+    else
+      render :show
```

Comment:
I don't believe that `driver.destroy` _can_ fail. If it did, this error handling code wouldn't work anyway - you don't have a `show` page for drivers.